packages/compose/src/higher-order/with-safe-timeout/index.tsx

```diff
@@ -72,8 +71,9 @@ const withSafeTimeout: PropInjectingHigherOrderComponent< TimeoutProps > = creat
 					...this.props,
 					setTimeout: this.setTimeout,
 					clearTimeout: this.clearTimeout,
-				};
-				return <OriginalComponent { ...( props as TProps ) } />;
+				} as TProps;
```

Comment:
Why do we need this type assertion?

Reply:
I'm not totally confident on my understand here, but I believe that it's because of the `Omit`. What I'm trying to convey is that the wrapped component doesn't want the timer functions to be passed in from the outside, but it still provides them to the wrapped component. I think there could be an issue here with React saying that `props` could contain any number of properties and TypeScript wants to complain that we might get ones we don't want or expect.

Reply:
> Why do we need this type assertion?

It's described in this TS issue: https://github.com/microsoft/TypeScript/issues/35858#issuecomment-573909154

Quote:

> TS isn't capable of the higher-order reasoning needed to understand that `Exclude<T, k> & { [k]: T[k] }` is equivalent to T.

If you observe how the type of `props` is `Omit<TProps, ...>` and how the value of local `props` value is constructed with `{ ...this.props, ...}`, it's exactly the same situation.